examples/CMakeModules/AddExample.cmake

```diff
@@ -0,0 +1,30 @@
+include(CMakeParseArguments)
+include(OSFlags)
+
+function(add_example target_name source backend)
```

Comment:
The CMake file should also be part of the example and you should try to simplify how to use the examples with with forge. I think you should only need 2 lines for this.